### PR TITLE
Add details to 3W page

### PR DESCRIPTION
**Pull request by barshathakuri**
Details
    - Fix latest PR comments
    - Fix UI inconsistencies
    
## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: src/App/routes.tsx [hunk 1]
```diff
@@ -240,6 +240,20 @@ const countryThreeW = customWrapRoute({
...
+const countryThreeWDetails = customWrapRoute({
+    path: 'three-w/:threeWId',
+    component: {
+        render: () => import('#views/ThreeWDetails'),
+        props: {},
+    },
+    parent: root,
+    wrapperComponent: Auth,
+    context: {
+        title: 'ThreeW Detail',
```
Comment (samshara) [inline src/App/routes.tsx]:
```suggestion
        title: 'ThreeW Details',
```

**Review by samshara — APPROVED** [review]

File: src/views/ThreeWDetails/i18n.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+    "namespace": "surge",
```
Comment (tnagorra) [inline src/views/ThreeWDetails/i18n.json]:
```suggestion
    "namespace": "threeWDetails",
```

File: src/views/ThreeWDetails/index.tsx [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+                                    : undefined}
+                            </>
+                        )}
+                    />
+                    {isDefined(projectResponse?.reporting_ns_contact_name) && (
+                        <>
+                            <TextOutput
+                                label={strings.nSContactTitle}
+                                value={
+                                    `${projectResponse?.reporting_ns_contact_name ?? ''}
```
Comment (tnagorra) [inline src/views/ThreeWDetails/index.tsx]: We do not need fallback for this variable

File: src/views/ThreeWDetails/index.tsx [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+                                    ,
+                                    ${projectResponse?.reporting_ns_contact_role ?? ''}`
```
Comment (tnagorra) [inline src/views/ThreeWDetails/index.tsx]: Let's filter undefined string and join by a `,`

**Review by tnagorra — DISMISSED** [review]

File: src/views/ThreeWDetails/index.tsx [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+    const { threeWId } = useParams<{ threeWId: string }>();
+
+    const {
+        response: projectResponse,
+    } = useRequest({
+        skip: isNotDefined(threeWId),
+        url: '/api/v2/project/{id}/',
+        pathVariables: {
+            id: Number(threeWId),
+        },
```
Comment (tnagorra) [inline src/views/ThreeWDetails/index.tsx]: Let's check if the threeWId is defined before we cast to number

File: src/views/ThreeWDetails/index.tsx [hunk 6]
```diff
@@ -0,0 +1,386 @@
...
+                        label={strings.lastModifiedOnTitle}
+                        value={
+                            resolveToComponent(strings.lastModifiedDetail, {
+                                date: (
+                                    <DateOutput
+                                        value={projectResponse?.modified_at}
+                                    />
+                                ),
+                                user: (
+                                    <>{projectResponse?.modified_by_detail}</>
```
Comment (tnagorra) [inline src/views/ThreeWDetails/index.tsx]: Let's remove the fragment here

File: src/views/ThreeWDetails/index.tsx [hunk 7]
```diff
@@ -0,0 +1,386 @@
...
+                                            {
+                                                countryId: projectResponse
+                                                    ?.project_country_detail.id,
+                                            },
+                                        ) : undefined}
+                                >
+                                    {projectResponse?.project_country_detail?.name}
+                                </Link>
+                                {projectResponse?.project_districts_detail
+                                    ? `${projectResponse?.project_districts_detail?.map((district) => district?.name).join(', ')}`
```
Comment (tnagorra) [inline src/views/ThreeWDetails/index.tsx]: Why reason to use string template here?

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/App/routes.tsx [hunk 8]
```diff
@@ -674,6 +674,20 @@ const newThreeWProject = customWrapRoute({
     },
 });
 
+const threeWProjectDetail = customWrapRoute({
```
Comment (AdityaKhatri) [inline src/App/routes.tsx:693]: Support this on listing page as well

File: src/views/ThreeWProjectDetail/index.tsx [hunk 9]
```diff
@@ -0,0 +1,448 @@
...
+                            date: (
+                                <DateOutput
+                                    value={projectResponse?.modified_at}
+                                />
+                            ),
+                            user: projectResponse?.modified_by_detail?.username,
+                        })
+                    }
+                    strongLabel
+                    withoutLabelColon
```
Comment (AdityaKhatri) [inline src/views/ThreeWProjectDetail/index.tsx]: modified_by_detail is null sometimes.
Don't display this is its empty.

File: src/views/ThreeWProjectDetail/index.tsx [hunk 10]
```diff
@@ -0,0 +1,448 @@
...
+                                        className={styles.countryLink}
+                                        withForwardIcon
+                                        to={projectResponse?.project_country_detail.id
+                                            ? generatePath(
+                                                countryRoute.absolutePath,
+                                                {
+                                                    countryId: projectResponse
+                                                        ?.project_country_detail.id,
+                                                },
+                                            ) : undefined}
```
Comment (AdityaKhatri) [inline src/views/ThreeWProjectDetail/index.tsx]: write this outside for simplicity and less nesting

File: src/views/ThreeWProjectDetail/index.tsx [hunk 11]
```diff
@@ -0,0 +1,448 @@
...
+                                    >
+                                        {projectResponse?.project_country_detail?.name}
+                                    </Link>
+                                    {projectResponse?.project_districts_detail
+                                        ? projectResponse?.project_districts_detail?.map((district) => district?.name).join(', ')
+                                        : null}
```
Comment (AdityaKhatri) [inline src/views/ThreeWProjectDetail/index.tsx]: write this outside for simplicity and less nesting

File: src/views/ThreeWProjectDetail/index.tsx [hunk 12]
```diff
@@ -0,0 +1,448 @@
...
+                            withoutLabelColon
+                        />
+                    </Container>
+                    <div className={styles.separator} />
+                    {projectResponse !== undefined
+                        && projectResponse?.annual_split_detail?.length > 0
+                        ? (
+                            <>
+                                {projectResponse?.annual_split_detail?.map((split) => (
+                                    <div className={styles.yearList}>
```
Comment (AdityaKhatri) [inline src/views/ThreeWProjectDetail/index.tsx]: UI is not consistent with the previous one.
image

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: src/App/routes.tsx (hunk 8 above)
Comment (barshathakuri) [reply src/App/routes.tsx:693]: Please explain there is another PR for list page and I'll add all route in that PR.

File: src/views/ThreeWProjectDetail/index.tsx [hunk 13]
```diff
@@ -0,0 +1,463 @@
...
+                                                label={strings.threeWTotal}
+                                                value={split.reached_total}
+                                                valueType="number"
+                                                strongValue
+                                                withoutLabelColon
+                                            />
+                                        </div>
+                                    </>
+                                ))}
+                            </div>
```
Comment (AdityaKhatri) [inline src/views/ThreeWProjectDetail/index.tsx]: Use list

**Pull request merged**